cranelift/codegen/src/isa/s390x/lower.isle

```diff
@@ -2133,229 +2135,165 @@
       (vec_unpacku_low $I32X4 y))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 13107)))
-      (if-let (imm8x16 _ _ 24 25 _ _ 26 27 _ _ 28 29 _ _ 30 31) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 _ _ 24 25 _ _ 26 27 _ _ 28 29 _ _ 30 31) 13107)))
       (vec_unpacku_low $I16X8 y))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 21845)))
-      (if-let (imm8x16 _ 24 _ 25 _ 26 _ 27 _ 28 _ 29 _ 30 _ 31) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 _ 24 _ 25 _ 26 _ 27 _ 28 _ 29 _ 30 _ 31) 21845)))
       (vec_unpacku_low $I8X16 y))
 
 ;; Special patterns that can be implemented via PERMUTE DOUBLEWORD IMMEDIATE.
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 0 1 2 3 4 5 6 7 24 25 26 27 28 29 30 31) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 0 1 2 3 4 5 6 7 24 25 26 27 28 29 30 31) 65535)))
       (vec_permute_dw_imm $I8X16 x 0 y 1))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23) 65535)))
       (vec_permute_dw_imm $I8X16 x 1 y 0))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 16 17 18 19 20 21 22 23 8 9 10 11 12 13 14 15) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 16 17 18 19 20 21 22 23 8 9 10 11 12 13 14 15) 65535)))
       (vec_permute_dw_imm $I8X16 y 0 x 1))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 24 25 26 27 28 29 30 31 0 1 2 3 4 5 6 7) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 24 25 26 27 28 29 30 31 0 1 2 3 4 5 6 7) 65535)))
       (vec_permute_dw_imm $I8X16 y 1 x 0))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15) 65535)))
       (vec_permute_dw_imm $I8X16 x 0 x 1))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 8 9 10 11 12 13 14 15 0 1 2 3 4 5 6 7) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 8 9 10 11 12 13 14 15 0 1 2 3 4 5 6 7) 65535)))
       (vec_permute_dw_imm $I8X16 x 1 x 0))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31) 65535)))
       (vec_permute_dw_imm $I8X16 y 0 y 1))
-(rule (lower (shuffle x y (shuffle_mask permute_mask 65535)))
-      (if-let (imm8x16 24 25 26 27 28 29 30 31 16 17 18 19 20 21 22 23) permute_mask)
+(rule (lower (shuffle x y (shuffle_mask (imm8x16 24 25 26 27 28 29 30 31 16 17 18 19 20 21 22 23) 65535)))
```

Comment:
For this whole group, removing the use of `if-let` resolved the priority naturally.